presto-orc/src/main/java/com/facebook/presto/orc/reader/MapStreamReader.java

```diff
@@ -100,6 +102,118 @@ public void startRowGroup(InputStreamSources dataStreamSources)
         currentReader.startRowGroup(dataStreamSources);
     }
 
+    @Override
+    public void setInputQualifyingSet(QualifyingSet qualifyingSet)
+    {
+        currentReader.setInputQualifyingSet(qualifyingSet);
+    }
+
+    @Override
+    public QualifyingSet getInputQualifyingSet()
+    {
+        return currentReader.getInputQualifyingSet();
+    }
+
+    @Override
+    public QualifyingSet getOutputQualifyingSet()
+    {
+        return currentReader.getOutputQualifyingSet();
+    }
+
+    @Override
+    public void setOutputQualifyingSet(QualifyingSet set)
+    {
+        currentReader.setOutputQualifyingSet(set);
+    }
+
+    @Override
+    public QualifyingSet getOrCreateOutputQualifyingSet()
+    {
+        return currentReader.getOrCreateOutputQualifyingSet();
+    }
+
+    @Override
+    public void setFilterAndChannel(Filter filter, int channel, int columnIndex, Type type)
+    {
+        directReader.setFilterAndChannel(filter, channel, columnIndex, type);
+    }
+
+    @Override
+    public int getChannel()
+    {
+        return directReader.getChannel();
+    }
+
+    @Override
+    public Block getBlock(int numFirstRows, boolean mayReuse)
+    {
+        return currentReader.getBlock(numFirstRows, mayReuse);
+    }
+
+    @Override
+    public Filter getFilter()
+    {
+        return directReader.getFilter();
+    }
+
+    @Override
+    public void erase(int end)
+    {
+        if (currentReader == null) {
+            return;
+        }
+        currentReader.erase(end);
+    }
+
+    @Override
+    public void compactValues(int[] positions, int base, int numPositions)
+    {
+        currentReader.compactValues(positions, base, numPositions);
+    }
+
+    @Override
+    public int getPosition()
+    {
+        return currentReader.getPosition();
+    }
+
+    @Override
+    public int getResultSizeInBytes()
+    {
+        if (currentReader == null) {
```

Comment:
Prior comment:

---
Some methods check if currentReader is null, but many assume it is not. Perhaps, add a comment explaining why the check is needed: // may be called before startStripe